File: lesson_22_java/src/main/java/com/lesson22java/controllers/CustomerController.java
package com.lesson22java.controllers;

import com.lesson22java.entities.Customer;
import com.lesson22java.services.CustomerService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@AllArgsConstructor
@RequestMapping("/customers")
public class CustomerController {
    private CustomerService customerService;

    @GetMapping("/create")
    public Customer create(@ModelAttribute Customer customer) {
        return customerService.create(customer);
    }

    @GetMapping("/getAll")
    public Iterable<Customer> getAll() {
        return customerService.getAll();
    }

    @GetMapping("/getOne/{id}")
    public Customer getOne(@PathVariable Long id) {
        return customerService.getOne(id);
    }

    @GetMapping("/edit")
    public Customer edit(@ModelAttribute Customer customer) {
        return customerService.edit(customer);
    }

    @GetMapping("/delete/{id}")
    public void delete(@PathVariable Long id) {
        customerService.delete(id);
    }
}

File: lesson_22/src/main/kotlin/com/lesson22/controllers/CustomerController.kt
package com.lesson22.controllers

import com.lesson22.entities.Customer
import com.lesson22.services.CustomerService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/customers")
class CustomerController (val customerService: CustomerService) {
    @GetMapping("/get-all")
    fun getAll() = customerService.getAll()

    @GetMapping("/create")
    fun create(@ModelAttribute customer: Customer) = customerService.create(customer)

    @GetMapping("/delete/{id}")
    fun delete(@PathVariable id: Long) = customerService.delete(id)

    @GetMapping("/edit")
    fun edit(@ModelAttribute customer: Customer) = customerService.edit(customer)
}